### PR TITLE
Add DEFAULT_FROM_EMAIL setting handling

### LINKED ISSUE #676: Settings does not respect `DEFAULT_FROM_EMAIL`
## Description

We recently tried setting the `DEFAULT_FROM_EMAIL`[^1] in the environment for the API, but the `settings.py` file doesn't make use of it. We need to add some logic to the `settings.py` file to allow the default to be overridden.

[^1]: https://docs.djangoproject.com/en/4.0/ref/settings/#std:setting-DEFAULT_FROM_EMAIL

## Additional context

Blocks the resolution of #506

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #676  by @AetherUnbound 

## Description

Adds the setting from the issue.

## Testing Instructions

:shrug: Not sure how to test this locally. You could set the environment variable then use `manage.py shell` to check if it is set.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/settings.py [hunk 1]
```diff
@@ -298,6 +298,7 @@
 EMAIL_HOST_PASSWORD = config("EMAIL_HOST_PASSWORD", default="")
 EMAIL_SUBJECT_PREFIX = "[noreply]"
 EMAIL_USE_TLS = True
+DEFAULT_FROM_EMAIL = config("DEFAULT_FROM_EMAIL", default="")
```
Comment (AetherUnbound) [inline api/catalog/settings.py:301]: The [Django default](https://docs.djangoproject.com/en/4.0/ref/settings/#default-from-email) is `webmaster@localhost`, did we want to mirror that here so it's more obvious when this isn't set?

Comment (sarayourfriend) [reply api/catalog/settings.py:301]: Personally I'd vote for this group of _required_ email settings to not have defaults unless some parent setting was turned off.

```python
if config("ENABLE_EMAIL", default=True, cast=bool):
    DEFAULT_FROM_EMAIL = config("DEFAULT_FROM_EMAIL")  # no default! This should be an error if you don't supply it
    # etc...
```

Comment (AetherUnbound) [reply api/catalog/settings.py:301]: I love this!

**Review by AetherUnbound — APPROVED** [review]
Thanks for jumping on this @sarayourfriend! This can be merged, but I like your suggestion about an `ENABLE_EMAIL` setting 💯

**Review by stacimc — APPROVED** [review]

**Pull request merged**